File: primes/primes.ts
const prime = (n: number): boolean => {
  if (n <= 1) {
    return false;
  }

  let i: number = 2;

  while (i * i <= n) {
    if (n % i === 0) {
      return false;
    }

    i += 1;
  }
  return true;
};

const num: number = 6;

if (!prime(num)) {
  console.log(`The number ${num} is not prime number`);
} else {
  console.log(`The number ${num} is a prime number`);
}

File: primes/primes.zig
const std = @import("std");
const print = std.debug.print;

pub fn primes(n: u32) bool {
    if (n <= 1) {
        return false;
    }

    var i: u32 = 2;

    while (i * i <= n) {
        if (n % i == 0) {
            return false;
        }
        i += 1;
    }

    return true;
}

pub fn main() anyerror!void {
    const num: u32 = 6;
    const res: bool = primes(num);
    if (!res) {
        print("The number {} is not prime number", .{num});
    } else {
        print("The number {} is a prime number", .{num});
    }
}
